python/langsmith/_internal/_background_thread.py

```diff
@@ -199,6 +231,19 @@ def keep_thread_active() -> bool:
     ):
         _tracing_thread_handle_batch(client, tracing_queue, next_batch, use_multipart)
 
+def tracing_control_thread_func_compress(client_ref: weakref.ref[Client]) -> None:
+    client = client_ref()
+    if client is None:
+        return
+
+    while True:
+        result = _tracing_thread_drain_compressed_buffer(client)
+        if result is not None:
+            time.sleep(0.150)  # Simulate call to backend
+        else:
+            time.sleep(0.1)  # Avoid busy-waiting if no data ready
```

Comment:
Definitely need to improve this to use timeouts similar to the other thread function, will start implementing today

Reply:
done